packages/angular_devkit/build_angular/plugins/webpack/analytics.ts

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { analytics } from '@angular-devkit/core';
+import {
+  Compiler,
+  Module,
+  Stats,
+  compilation,
+} from 'webpack';
+import { Source } from 'webpack-sources';
+import Compilation = compilation.Compilation;
+
+const NormalModule = require('webpack/lib/NormalModule');
+
+interface NormalModule extends Module {
+  _source?: Source | null;
+  resource?: string;
+}
+
+const webpackAllErrorMessageRe = /^([^(]+)\(\d+,\d\): (.*)$/gm;
+const webpackTsErrorMessageRe = /^[^(]+\(\d+,\d\): error (TS\d+):/;
+
+/**
+ * Faster than using a RegExp, so we use this to count occurences in source code.
+ * @param source The source to look into.
+ * @param match The match string to look for.
+ * @param wordBreak Whether to check for word break before and after a match was found.
+ * @return The number of matches found.
+ * @private
+ */
+export function countOccurrences(source: string, match: string, wordBreak = false): number {
```

Comment:
would be nice to unit test this - in case we find some weird logging edge case

Reply:
Done (and fixed a bug)